### DISCUSSION TITLE
Proactive Synchronization

### DISCUSSION THREAD

**Discussion by ll13j1209** [afs/rdf-delta#223] [Q&A] [answered]
It seems like that if we have multiple delta fuseki server, the individual fuseki will not fetch the newest updates from the patch log server until someone make a request to it. For example, if I have a load balancer and 3 delta fuseki servers behind it, when I make a update request to one of the servers, the other two servers will not pick up the update unless there is a sparql request route to themselves. When this happens, the querying will be very slow since the fuseki need to catch up the newest update. 
Is there a way to remitigate the issue and let all the nodes to proactively catch up the update from the patch log server? I am thinking have a cronjob that constantly query to the individual server but I dont know if that will affect the performance. Thank you !

**Comment by afs** [comment]
Sending a harmless query will work. Something like `ASK{}`. The performance impact of that query is small (TDB2), and unless the server is under a near-continuous query load, probably negligible.  An actual update is more expensive than an `ASK{}`.

**Reply by ll13j1209** [reply]
Does patch log server expose any API that allow user to check the current version of patch? This way we only need to send a small query when there is an update instead of constantly making calls to the server.

**Reply by afs** [reply, marked as answer]
Check the current version - yes - but that isn't enough to determine whether an update is going to happen. That needs to know the state of the Fuseki server - it may be several versions behind.  Pinging it with `ASK{}` does that calculation and is cheap. If, say, once every 5 seconds, it should be negligible.

If you do want to avoid the regular ping, have the update request touch each server after the update returns.

**Reply by ll13j1209** [reply]
Great! That makes sense, thank you

**Reply by ll13j1209** [reply]
BTW, out of curiosity, what is the endpoint that patch log server expose that allow user to check the current version?

**Comment by afs** [comment]
Interface `DeltaLink` is everything : remote is `DeltaLinkHTTP`

All operations except patch push/get (called `append` and `fetch`) are via a (very lightweight) RPC-ish mechanism on `/$/rpc`.
